examples/robot_localization_failure/run.py

```diff
@@ -1,103 +1,311 @@
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.z",
+                                    ).alias("orientation.z"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.w",
+                                    ).alias("orientation.w"),
+                                ],
+                            ).alias("pose"),
+                        ],
+                    ).alias("/amcl_pose"),
+                    pl.col("is_delocalized"),
+                )
+            )
+            # Split the data into train and test
+            data = DataFrame(data)
+            train, test = TrainTestSplit(ratio=0.8, shuffle=True).split(data)
+            train_images = FeatureImagesData(
+                train.data.collect(),
+                image_size=config.architecture.image_size,
+                width_meters=config.architecture.width_meters,
+            )
+            train_datasets.append(train_images)
+            test_images = FeatureImagesData(
+                test.data.collect(),
+                image_size=config.architecture.image_size,
+                width_meters=config.architecture.width_meters,
+            )
+            test_datasets.append(test_images)
+        except Exception as e:
+            msg = f"Error loading {parquet_path}: {e}"
+            logger.exception(msg)
+            continue
+
+    # Combinedatasets
+    train_data = pl.concat([train.data for train in train_datasets])
+    test_data = pl.concat([test.data for test in test_datasets])
+
+    # Debug: Print columns to verify structure
+    logger.info(f"Train data columns: {train_data.columns}")
+    logger.info(f"Test data columns: {test_data.columns}")
+
+    # Create and train the learner
+    learner = ImageBasedLightningLearner(
+        module=CNN(
+            image_size=config.architecture.image_size,
+            in_channels=3,
+            learning_rate=config.learning.learning_rate,
+        ),
+        batch_size=config.learning.batch_size,
+        max_epochs=config.learning.epochs,
+        dataset_factory=lambda data: FeatureImagesData(
+            data,
+            image_size=config.architecture.image_size,
+            width_meters=config.architecture.width_meters,
+        ),
+        dataset_kwargs={
+            "image_size": config.architecture.image_size,
+            "width_meters": config.architecture.width_meters,
+        },
     )
-    | Drop("/scan", "/particle_cloud", "/momo/pose", "/amcl_pose")
-    # detect experiment slice points based on delocalization events
-    | DetectDelocalizations("/delocalizations", name="slice_points")
-    | Drop("/delocalizations")
-    | SliceTimeSeries(
-        time_series="measurements",
-        slice_points="slice_points",
+    model = learner.learn(
+        inputs=train_data.drop(["is_delocalized"]),
+        outputs=train_data.select(["is_delocalized"]),
```

Comment:
we should be able to use the flowcean strategies here. `learn_offline` for example?